### PR TITLE
Implement RPC-GET-V1-08

### PR DESCRIPTION
**Pull request by AkhilaIlla**

### CODE REVIEW

File: packages/rulesets/src/spectral/functions/utils.ts [hunk 1]
```diff
@@ -152,25 +152,47 @@ export function isXmsResource(schema: any) {
...
+  const index = path.lastIndexOf("/providers/")
+  if (index === -1) {
+    return []
+  }
+  const lastProvider = path.substr(index)
+  const result = []
+  const matches = lastProvider.match(SpecificResourcePathRegEx)
+  if (matches && matches.length) {
+    const match = matches[0]
+    const segments = match.split("/").slice(3)
```
Comment (bdefoy) [inline packages/rulesets/src/spectral/functions/utils.ts:189]: also could rename this to something like `resourcePathSegments`

File: packages/rulesets/src/spectral/functions/utils.ts [hunk 2]
```diff
@@ -152,25 +152,47 @@ export function isXmsResource(schema: any) {
...
+          return true
         }
       }
     }
   }
-  return false;
+  return false
+}
+
+const SpecificResourcePathRegEx = new RegExp("/providers/[^/]+(?:/\\w+/default|/\\w+/{[^/]+})+(?:\\?\\w+)?$", "gi")
```
[the remark this answers is not in the document]
Comment (AkhilaIlla) [reply packages/rulesets/src/spectral/functions/utils.ts:178]: Agreed, updated.
and thanks this is very helpful.

File: packages/rulesets/src/spectral/functions/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (AkhilaIlla) [reply packages/rulesets/src/spectral/functions/utils.ts:189]: as we split by "/"
for "/providers/Microsoft.Music/songs/{unstoppable}/artist/{sia}"
match.split("/") will be 
match[0] = ""
match[1] = "providers"
match[2] = Microsoft.Music""

so we select from, RT name

File: packages/rulesets/src/spectral/functions/utils.ts [hunk 3]
```diff
@@ -152,25 +152,52 @@ export function isXmsResource(schema: any) {
...
+          return true
         }
       }
     }
   }
-  return false;
+  return false
+}
+
+//const SingleResourceInstancePathRegEx = new RegExp("/providers/[^/]+(?:/\\w+/default|/\\w+/{[^/]+})+(?:\\?\\w+)?$", "gi")
```
Comment (bdefoy) [inline packages/rulesets/src/spectral/functions/utils.ts:178]: remove commented code

**Review by bdefoy — APPROVED** [review]

File: packages/rulesets/src/spectral/functions/utils.ts (hunk 3 above)
Comment (bdefoy) [inline packages/rulesets/src/spectral/functions/utils.ts:178]:
```suggestion
```

**Review by bdefoy — APPROVED** [review]

**Pull request merged**